src/api/index.ts

```diff
@@ -54,16 +69,19 @@ function createApp() {
     asyncHandler(async (req, res) => {
       const parsedRequest = await parseRequest(req, getHooksRequestSchema)
       const { address } = parsedRequest.query
-      const networkId =
-        'network' in parsedRequest.query
-          ? legacyNetworkToNetworkId[parsedRequest.query.network]
-          : parsedRequest.query.networkId
-      const positions = await getPositions(
-        networkId,
-        address,
-        config.POSITION_IDS,
-        config.GET_TOKENS_INFO_URL,
-      )
+      const networkIds = getNetworkIds(parsedRequest.query)
+      const positions = (
+        await Promise.all(
```

Comment:
Will this change the wallet UX (compared to making one request per network)?  What about the example where (for whatever reason) `getPositions` is broken for eth?  Would the one-request-per-network approach render positions for the other chains and just exclude eth ones (and the change here would cause the wallet to show no positions).

Reply:
...if we think the one-request-per-network UX is better, we could consider tweaking this to PR to return whatever positions are working

Reply:
Yes, if the hooks service fails to get positions for one network, the entire request would fail with my PR as written. As you've noted, if we did one request per network, we could handle failures differently on the client side, for instance by displaying just the positions which are successfully returned. I think that's the main downside with this approach. However, I thought the benefit of having only one network request for an arbitrary number of networks was worth it. My guess is that random network failures or latency would increase with the number of requests, so I'm suggesting 1-request-for-n-networks approach instead of 1-request-per-network

Reply:
And yeah, errors for just one network could be swallowed on the backend too if that's the experience we want to support on the client

Reply:
Thanks, yeah this raises an interesting question on what we want to do moving forward.
For blockchain-api we did 1 request per network on the client side. I think I was initially more in favor of a single request for all networks, but considering failures and displaying data to the user more quickly, I'm leaning more toward letting the client do N requests.
Especially here in this project where querying all positions could take longer time, as we add more apps on Ethereum.
Maybe something to discuss at tomorrow's eng meeting?